railties/test/commands/command_test.rb

```diff
@@ -0,0 +1,56 @@
+require_relative '../isolation/abstract_unit'
+require_relative '../env_helpers'
+require_relative '../../lib/rails/commands/command'
+
+class Rails::CommandTest < ActiveSupport::TestCase
+  include ActiveSupport::Testing::Isolation, EnvHelpers
+
+  DEFAULT_TASK_NAME = 'command:test'
+
+  def setup
+    build_app
+    create_command(task: DEFAULT_TASK_NAME)
+  end
+
+  def teardown
+    teardown_app
+  end
+
+  def test_command_creation
+    assert_match %r{Rails command defined successfully.}, run_command
+  end
+
+  def test_command_failure
+    assert_match %r{}, run_command(task: 'command:exception')
+  end
+
+  def test_command_non_existence
+    assert_match %r{Error: Command 'does:not:exist' not recognized}, run_command(task: 'does:not:exist')
+  end
+
+  def test_help_options
+    assert_match %r{Show this help.}, run_command(task: "#{DEFAULT_TASK_NAME} -h")
+  end
+
+  private
+
+    def run_command(task: DEFAULT_TASK_NAME)
+      Dir.chdir(app_path) { `bin/rails #{task}` }
+    end
+
+    def create_command(class_name: :test, task:)
```

Comment:
Just that we are exercising the command running code a bit. But then again I guess any behavior is tested through the commands we ship with.

Alright, if all the Railties tests pass, then it's fine. I'll look into what other tests need after this then 😁